src/odemis/driver/tfsbc.py

```diff
@@ -0,0 +1,277 @@
+        if serialnum:
+            for port in names:
+                if serialnum in port.hwid:
+                    self._serial = self._openSerialPort(port)
+                    return port  # Found it!
+            else:
+                raise ValueError("Couldn't find device with serial number %s in list of ports %s." % (serialnum, names))
+        else:
+            if len(names) == 1:
+                port = names[0]
+                self._serial = self._openSerialPort(port)
+                return port
+            elif len(names) > 1:
+                raise ValueError("Multiple ports detected. Please specify a serial number.")
+            else:
+                raise ValueError("Couldn't find port with pattern '%s'." % ports)
+
+    def _openSerialPort(self, port):
+        if self._port == "/dev/fake":
+            return BeamControllerSimulator()
+        else:
+            return ModbusSerialClient(method='rtu', port=port,
+                                      baudrate=BAUDRATE, timeout=TIMEOUT,
+                                      stopbits=STOPBITS, parity=PARITY,
+                                      bytesize=BYTESIZE)
+
+    def _setShift(self, value):
+        """
+        :param value (float, float): x, y shift from the center (in m)
+        """
+        gain = self._metadata.get(model.MD_CALIB_SCALE)
+        if not gain:
+            raise ValueError("No ampere-to-meter conversion factor specified. Please run the alignment procedure "
+                             "first.")
+        offset = self._metadata.get(model.MD_CALIB_TRANSLATION)
+        if not offset:
+            logging.warning("No deflection offset specified, assuming 0 amps.")
+            offset = (0, 0)
+
+        rotation = self._metadata.get(model.MD_CALIB_ROTATION)
+        if not offset:
+            logging.warning("No rotation specified, assuming 0 rad.")
+            rotation = 0
+
+        # Convert to beam coordinate system (rotation)
+        value = (value[0] * math.cos(rotation) + value[1] * math.sin(rotation),
+                 -value[0] * math.sin(rotation) + value[1] * math.cos(rotation))
+
+        # Convert to numpy.array to simplify calculation
+        value = numpy.array(value, dtype=numpy.float64)
+        gain = numpy.array(gain, dtype=numpy.float64)  # amps / m
+        offset = numpy.array(offset, dtype=numpy.float64)  # amps
+        amp_range = numpy.array(RANGE_AMPS, dtype=numpy.float64)
+
+        # Convert to amperes (with gain)
+        value *= gain
+
+        # Add offset
+        value -= offset  # subtract offset (center amplitude)
+
+        # Convert to 16-bit integer
+        conversion_factor = 0xFFFF / (amp_range[1] - amp_range[0])
+        valuePos = (value - amp_range[0]) * conversion_factor
+        valueNeg = (-value - amp_range[0]) * conversion_factor
+        # Note: in the example script a constant of 0.5 is added before converting
+        # to int to always round up (why?)
```

Comment:
No, adding a constant of 0.5 doesn't always round *up*, it rounds to the closest int (as C int() rounds down). In Python there is a function for that, cleverly named `round()`.
I'd suggest to use it too. Note that you still need `int` to force it being an int: `int(round(x))`.